datamol/align.py

```diff
@@ -0,0 +1,252 @@
+from typing import Optional
+from typing import Union
+from typing import Iterable
+from typing import Sequence
+
+from collections import defaultdict as ddict
+
+from rdkit.Chem import rdDepictor
+from rdkit.Chem import rdMolAlign
+
+import pandas as pd
+import datamol as dm
+
+
+from .types import Mol
+
+
+def should_align(mols: Iterable[Mol]) -> bool:
+    """Check whether a list of molecules should be aligned by checking
+    if at least one of them has computed coordinates.
+
+    Args:
+        mols: List of molecules.
+    """
+    return not any(mol.GetNumConformers() > 0 for mol in mols)
+
+
+def compute_2d_coords(mol: Mol, copy: bool = True, verbose: bool = False) -> Mol:
+    """Compute 2D coordinates for a molecule.
+
+    Args:
+        mol: A molecule.
+        copy: Whether to copy the molecule.
+    """
+    if copy:
+        mol = dm.copy_mol(mol)
+
+    with dm.without_rdkit_log(enable=not verbose):
+        rdDepictor.Compute2DCoords(mol)
```

Comment:
Could't properly debug this but using `pytest` I have a segmentation fault error using auto-align with `strip-scaffold` only. I can reproduce this outside of pytest...

Reply:
I haven't checked. I noticed some rdkit warnings when using strip-scaffold. This is why I have added the log decorator.

It only fails with pytest (making me think it's gonna be tricky to debug).

Here is the code to reproduce:

```python
data: pd.DataFrame = dm.cdk2(as_df=True)  # type: ignore
data = data.iloc[:32].copy()  # type: ignore

data["aligned|_mol"] = dm.align.auto_align_many(data["mol"], partition_method="strip-scaffold")
```

Reply:
Not sure. Didn't have any issue either. Will revisit later.

Reply:
I think it's just that sometime `rdDepictor.Compute2DCoords` fails silently which is fine and what we want but somehow when in a multithreading mode (`pytest`), there is a weird rdkit segmentation fault being triggered. I don't think we'll get that issue using multiprocessing or joblib.

I am fine just ignoring it for now, and we can address it later if it becomes problematic.

Reply:
For the test, switching to `dm.solubility` instead of cdk2 does the job.